### DISCUSSION TITLE
CVE-2018-11087 Bug Report

### DISCUSSION THREAD

**Discussion by VP-66** [spring-projects/spring-amqp#2782] [Q&A] [answered]
Hi Team, 

I am doing some research into the CVE-2018-11087 and looking for the bug report that first identified/mentioned this issue. However, after doing a lot of research, I was not able to find this bug report/issue in this repository. 

Any guidance/help as to where I could find this information would be highly helpful!

I have found the fixing commit, however, which is: https://github.com/spring-projects/spring-amqp/commit/aff4d0aefcdb99726fd739abf3b9bb96df97b0f

The jira task mentioned in the commit also does not work (ie, no redirection to relevant issue: https://jira.spring.io/browse/AMQP-830) 

Any help would be great, 
Thanks!

**Comment by artembilan** [comment]
It is very old CVE though, therefore indeed something might be lost.
On the other hand it is better to not report bug for security vulnerability into the public. It can definitely cause a demage to those affectected, but without an attack yet. Plus it is very good practice to hide exploit details as much as possible. So, I’m not sure what is your goal, but that commit should be enough for community to know that fix is there.

**Reply by VP-66** [reply]
Thanks for your reply @artembilan !

I understand the benefits for not reporting the bug to the public as it can easily be exploited then. 

The sole purpose of this task is for a university project related to the processing of identifying and reporting CVE in open-source projects. As for this, we are required to find the corresponding bug report for the fixing commit. 

Thanks for your help !

**Reply by artembilan** [reply]
Right, as I said: with time and evolution something might be lost. We have migrated from JIRA while back, but here is related migrated GH issue: https://github.com/spring-projects/spring-amqp/issues/2373.

Also as we agreed , you can see that not too much info in the issue. Just because it is a security vulnerability. I just made a search for that JIRA ticket number. So, in fact it is not lost 😅.

**Reply by VP-66** [reply]
That is really helpful, @artembilan !

That is the same GH issue I had come across whilst doing my research - just one last thing would be great to confirm is that the JIRA ticket is not visible to the public, correct? If I am incorrect, is there any way I can search up the JIRA ticket number from my end?

Again, thanks a lot for your guidance :)

**Reply by artembilan** [reply]
The JIRA is not there for the reason you can see when you follow its link. Im not sure why you are eager for that JIRA, but GH issues have same meaning as JIRA. Plus that ticket has had exactly the same content you see in that GH issue. 
What is your goal here, please?

**Comment by VP-66** [comment]
Hi @artembilan , that is all good I just wanted to see if there was any discussion related to how the bug was found/fixed - including any sort of suggestions. 

That is all good though, thanks a lot for all your help. Really appreciated :)

**Reply by artembilan** [reply, marked as answer]
“How the bug was found” is 100% exploit and that is not what I would recommend to share with anyone. “How it was fixed” is also some kind of signal for hackers to crack all those who have not updated yet. So, that is also something we try to keep as obfuscated as possible.
 Therefore don’t try to find clues in this project about origins of this or that CVE.

**Reply by VP-66** [reply]
@artembilan that makes so much sense, thanks a lot!!

**Comment by acogoluegnes** [comment]
I don't remember the exact origin of the CVE, but here is some extra information:

* the related GH issue for the RabbitMQ AMQP 091 Java client: https://github.com/rabbitmq/rabbitmq-java-client/issues/394
* the commit that fixed the issue: https://github.com/rabbitmq/rabbitmq-java-client/commit/fcc3dbb5f797e7e340b6ddc5c2faa2fc77a1c905

The idea was to make hostname verification easier to activate.
